contracts/src/bridge/L1/ObscuroBridge.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./IObscuroBridgeAdmin.sol";
+import "../IBridge.sol";
+import "../ITokenFactory.sol";
+import "../../messaging/messenger/CrossChainEnabledObscuro.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+// This is the Ethereum side of the Obscuro Bridge.
+// End-users can interact with it to transfer ERC20 tokens and native eth to the Layer 2 Obscuro.
+contract ObscuroBridge is
+    CrossChainEnabledObscuro,
+    IBridge,
+    IObscuroBridgeAdmin,
+    AccessControl
+{
+    // This is the role that is given to the address that represents a native currency
+    bytes32 public constant NATIVE_TOKEN_ROLE = keccak256("NATIVE_TOKEN");
+
+    // This is the role that is given to addresses which are ERC20 contract.
+    // If we have assigned a role to a contract it is considered whitelisted.
+    bytes32 public constant ERC20_TOKEN_ROLE = keccak256("ERC20_TOKEN");
+
+    // This is the role of the address that can perform administrative changes
+    // like adding or removing tokens.
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+
+    address remoteBridgeAddress;
+
+    constructor(address messenger) CrossChainEnabledObscuro(messenger) {
+        _grantRole(ADMIN_ROLE, msg.sender);
+        _grantRole(NATIVE_TOKEN_ROLE, address(0x0));
```

Comment:
Native currency is the empty address ?

Reply:
This bit is unfinished as mentioned in the sprint planning, I'm planning to have a gateway contract or something there.